src/contexts/NameListContext.js

```diff
@@ -125,6 +128,17 @@ class NameListProvider extends React.Component {
     this.setState({ namesPerPage: number });
     let results = [...this.state.list];
     this.setState({ listView: results.splice(0, number) });
+    this.onPageChange(this.state.page);
+  };
+
+  // Change Page
+  onPageChange = (newPageNumber) => {
+    let pageStart = (newPageNumber - 1) * this.state.namesPerPage;
```

Comment:
How does this function behave if `newPageNumber` is too small or too large?

Reply:
I have it to conditionally render the previous button to not appear on the first page. Just added a new function to not go over, but gonna try and make it look more elegant...